annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastServiceCreator.java

```diff
@@ -0,0 +1,59 @@
+package com.exactpro.epfast.annotation.processing;
+
+import javax.annotation.processing.Filer;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.Element;
+import javax.tools.FileObject;
+import javax.tools.StandardLocation;
+import java.io.*;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class FastServiceCreator {
+    private Filer filer;
+
+    public FastServiceCreator(ProcessingEnvironment processingEnvironment) {
+        this.filer = processingEnvironment.getFiler();
+    }
+
+    public void updateServiceEntries(TypeName service, HashSet<TypeName> providers,
```

Comment:
HashSet -> Set. It's public method and it's better don't require any particular Set implementation.